### PR TITLE
update variable documentation

### LINKED ISSUE #1674: Adapt documentation to new variable concept
* Update the data flow sections
* Document the new flag from https://github.com/zeebe-io/zeebe/issues/1610
* ...

### PR DESCRIPTION
**Pull request by saig0**
closes #1674

### CODE REVIEW

File: docs/src/reference/variables.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+target: sender.iban</pre></td>
+    <td><pre>
+sender: {"name": "John",
+  "iban": "DE456"}</pre></td>
+  </tr>
+</table>
+
+### Output Mappings
+
+Output mappings can be used to customize how job/message variables are merged into the workflow instance. They can be defined on service tasks, receive tasks and message catch events.
```
Comment (Zelldon) [inline docs/src/reference/variables.md]:
```suggestion
Output mappings can be used to customize how job/message variables are merged into the workflow instance. They can be defined on service tasks, receive tasks, sub processes and message catch events.
```

File: docs/src/yaml-workflows/data-flow.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Data Flow
```
Comment (Zelldon) [inline docs/src/yaml-workflows/data-flow.md]: Please update the control flow side as well

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
I like it :+1: just few small suggestions

**Comment by saig0** [comment]
@Zelldon , I applied the review hints. Please have a look at the latest commit.

**Review by ChrisKujawa — APPROVED** [review]
bors r+

**Pull request merged**